src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,55 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int totalSupply = 0;
+        int totalBuy = 0;
+
+        try {
+            totalSupply = readData(fromFileName, SUPPLY);
+            totalBuy = readData(fromFileName, BUY);
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading data from file: " + fromFileName, e);
+        }
+
+        int result = totalSupply - totalBuy;
+
+        try (FileWriter writer = new FileWriter(toFileName)) {
+            writer.write(SUPPLY + "," + totalSupply + System.lineSeparator());
+            writer.write(BUY + "," + totalBuy + System.lineSeparator());
+            writer.write("result," + result + System.lineSeparator());
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing data to file: " + toFileName, e);
+        }
```

Comment:
The catch block here is a good example of how to handle exceptions. It catches the IOException, adds a custom error message, and wraps it in a RuntimeException. This is a good practice because it provides more information about the error and doesn't require the calling code to handle the IOException.